dm/worker/worker.go

```diff
@@ -190,6 +170,114 @@ func (w *Worker) Close() {
 	w.l.Info("Stop worker")
 }
 
+// EnableRelay enables the functionality of start/watch/handle relay
+func (w *Worker) EnableRelay() error {
+	// 1. adjust relay starting position, to the earliest of subtasks
+	_, subTaskCfgs, _, err := ha.GetSubTaskStageConfig(w.etcdClient, w.cfg.SourceID)
+	if err != nil {
+		return err
+	}
+	if err = copyConfigFromSourceForEach(subTaskCfgs, w.cfg); err != nil {
+		return err
+	}
```

Comment:
How about combine them with L246-253

Reply:
OK, and I found in https://github.com/pingcap/dm/blob/8157e33bd21d85504ba533d7ab5dc1cb26417823/dm/worker/worker.go#L301-L309
we forget to `copyConfigFromSourceForEach`.

Reply:
https://github.com/pingcap/dm/pull/1507/commits/c151ea0a21a0cabca26aa272a97821f00a4858d9